### ISSUE TITLE
Albums from same artists not displayed in alphabetical order

### ISSUE THREAD

**Issue by dassencio** [Rello/audioplayer#80]
### Expected behavior
On the main Audio Player page, where all albums are visualized, the albums must be displayed sorted first by artist names, and albums belonging to the same artist should be displayed ordered by the album names.

### Actual behavior
The albums are indeed ordered by artist names, but the albums belonging to a given artist are not ordered correctly. For instance, under "Café del Mar", the albums "Volume 6", "Volume 8" and "Volume9" do not appear in this order, but as "Volume 6", "Volume 9" and "Volume 8". This makes it difficult to find the desired album for artists which have many albums in the collection.

### Steps to reproduce the behavior
1. 
2. 
3. 

### Server configuration
**Operating system**:
Linux/Windows

**Web server:**

**Database:**
MySQL

**PHP version:**
7

**ownCloud/Nextcloud version:** (see `/status.php`)
9.1.1

**Updated from an older ownCloud/Nextcloud or fresh install:**
Updated (but problem was always there)

**Where did you install Audio Player from:**

**Are you using external storage, if yes which one:** `local/smb/sftp/...`

**Are you using encryption:** yes/no
yes

### Client configuration
**Operating system:**
Linux/Windows

**Browser:**
Firefox/Chromium

#### ownCloud/Nextcloud log
<details>
<summary>ownCloud/Nextcloud log (`/data/[owncloud|nextcloud].log`)</summary>
```
Insert your ownCloud/Nextcloud log here
```
</details>

#### Web server error log (optional)
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Browser log (optional)
<details>
<summary>Browser log</summary>
```
Insert your browser log here
```
</details>

**Comment by Rello** [comment]
Thank you for the advice!
I will check and include it in the 1.3 which is being finalized...

**Comment by Rello** [comment]
Hello,
I just checked the code. The Albums are sorted ASC by artist name.
https://github.com/Rello/audioplayer/blob/master/controller/musiccontroller.php#L204 

I will check how to sort by artist & album

**Comment by dassencio** [comment]
Hi @Rello,

I can see that `sortAlbums` is only used once in the code. In this case, there is no need to keep it so general, i.e., there is no need to allow the user to specify the `$sortAttribute`. My suggestion is as follows (please notice that I'm not a php specialist):

``` php
     public static function sortAlbums($files, $sortDescending = false) {
        $sortFunc = 'compareAlbumNames';
        usort($files, array('\OCA\audioplayer\Controller\MusicController', $sortFunc));
        if ($sortDescending) {
            $files = array_reverse($files);
        }
        return $files;
    }
```

and then (I'm not sure if this is even valid php code, but I think you will get the idea):

``` php
   public static function compareAlbumNames($a, $b) {
               if ($a[self::artist] === $b[self::artist])
               {
                     return \OCP\Util::naturalSortCompare($a[self::name], $b[self::name]);
               }
              return \OCP\Util::naturalSortCompare($a[self::artist], $b[self::artist]);
    }
```

This will sort albums by artist first and also also sort albums from a given artist alphabetically.

**Comment by Rello** [comment]
Hello,
Thank you for the feedback. This is still legacy code which I need to double check. 
I think it should be done in a much simpler and straight forward way.
I will check

**Comment by Rello** [comment]
Hello @dassencio 
issue was solved by a much cleaner array_multisort.
will be part of 1.3 coming soon.

thank you for your support
![screen shot 2016-11-11 at 22 38 01](https://cloud.githubusercontent.com/assets/13385119/20231069/8b64df4a-a85f-11e6-8fdd-9947964c058c.png)

```
    private function sortArrayByFields($data)
    {
        foreach ($data as $key => $row) {
            $first[$key] = $row['artist'];
            $second[$key] = $row['name'];
        }
        array_multisort($first, SORT_ASC, SORT_STRING,$second , SORT_ASC, SORT_STRING, $data);
        return $data;
    }
```

**Comment by dassencio** [comment]
@Rello thank you very much for fixing this. When will the fix become available? Only on the next ownCloud update, or do we have to somehow update manually?

Thanks!

**Comment by Rello** [comment]
Hello,
It will be in the next Audio Player 1.3 release. 
I hope to have it published in the Appstores (OC & NC) beginning of the week

**Comment by kernel1024** [comment]
Just small feature request. It would be cool, if sort field can be configured from gui for albums list.

**Comment by ghost** [comment]
@kernel1024 This issue is closed with the release of v1.3.0, as you can see. Please open a new issue. Thanks.

**Comment by Rello** [comment]
sorting was adjusted with 1.4 to sort case insensitive
`|SORT_FLAG_CASE`
